server/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTests.java

```diff
@@ -97,8 +99,11 @@ protected TermsQueryBuilder doCreateTestQueryBuilder() {
     }
 
     private TermsLookup randomTermsLookup() {
-        return new TermsLookup(randomAlphaOfLength(10), randomAlphaOfLength(10), randomAlphaOfLength(10),
-                termsPath).routing(randomBoolean() ? randomAlphaOfLength(10) : null);
+        TermsLookup lookup = randomBoolean()
+            ? new TermsLookup(randomAlphaOfLength(10), randomAlphaOfLength(10), termsPath)
+            : new TermsLookup(randomAlphaOfLength(10), randomAlphaOfLength(10), randomAlphaOfLength(10), termsPath);
```

Comment:
nit: add comment to understand why these two cases are necesarry. It makes sense in the context of this PR but I doubt someone will make sense of it in two months from now. Maybe also name the "type" argument specifically just to make sure the two cases here are related to types removal to make further refactoring easier.

Reply:
👍